widget/embedded/src/pages/CustomTokensPage.tsx

```diff
@@ -0,0 +1,165 @@
+import type { Token } from 'rango-sdk';
+
+import { i18n } from '@lingui/core';
+import {
+  Button,
+  CustomTokensDarkIcon,
+  CustomTokensIcon,
+  DeleteIcon,
+  Divider,
+  IconButton,
+  MessageBox,
+  NotFound,
+  styled,
+} from '@rango-dev/ui';
+import React, { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import { WatermarkedModal } from '../components/common/WatermarkedModal';
+import { Layout, PageContainer } from '../components/Layout';
+import { SearchInput } from '../components/SearchInput';
+import { TokenList } from '../components/TokenList';
+import { navigationRoutes } from '../constants/navigationRoutes';
+import { useTheme } from '../hooks/useTheme';
+import { useAppStore } from '../store/AppStore';
+import { containsText, getContainer } from '../utils/common';
+
+const Content = styled('div', {
+  display: 'flex',
+  justifyContent: 'space-between',
+  flexDirection: 'column',
+  flex: 1,
+});
+const NotFoundContent = styled('div', {
+  display: 'flex',
+  flexDirection: 'column',
+  alignItems: 'center',
+  justifyContent: 'center',
+  flex: '0.75',
+});
+const DeleteIconButton = styled(IconButton, {
+  '&:hover': {
+    '& svg': {
+      color: '$secondary550',
+    },
+  },
+});
+export function CustomTokensPage() {
+  const [searchedFor, setSearchedFor] = useState<string>('');
+  const { customTokens, deleteCustomToken } = useAppStore();
+  const { isDarkTheme } = useTheme({});
+  const navigate = useNavigate();
+  const [isOpenDeleteModal, setIsOpenDeleteModal] = useState<boolean>(false);
+  const [selectedToken, setSelectedToken] = useState<Token>();
+  const searchHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value;
+    setSearchedFor(value);
+  };
+
+  const filterCustomTokens = () =>
```

Comment:
There is no point to make this lazy, because you are calling it in your render always.

Reply:
I think it's not needed to use `useMemo` here since `searchedFor` and `customTokens` are the main values of this page which means they will cause rerender in most cases. And also custom tokens wouldn't be a long list I guess, so memoizing the computation wouldn't help us to much. I prefer to keep the code simple and use this types of optimization only if needed.

Reply:
`filter` also is a verb which can be used for actions/functions. here you are keeping result of your filter. I suggest `filteredCustomTokens` or a more contextual name to show search functionality like `customTokensResults`. I prefer the latter one.

Reply:
There is also a `console` at line 69. please remove.

Reply:
Okay, I changed the variable name to ### customTokensResults and returned the function to simple code. I also removed the console.